### DISCUSSION TITLE
Humax FoxsatHDR remote not working

### DISCUSSION THREAD

**Discussion by gbjbasdw** [GerryDazoo/Slinger#440] [Q&A]
I have come to a dead end trying to get remote control on my Humax FoxsatHDR Freeesat recorder and wonder if anyone can suggest where to go next.

I have Slinger working just fine with video and audio from the Humax but no matter what code I send nothing is happening. I've tested the IR blaster just in case that had stopped working but it works fine on my Tivo box,

I've tried Slingplayer 1.5 ( which I used to unbrick the ProHD box ) to load stuff into the ProHD but there is no option for Brand Humax FoxsatHDR only a straight Humax PVR (which I am currently using) . I found a SB0312.bin file which is supposed to be for the Humax box on the net but that couldn't be loaded in Slingplayer 1.5.

From memory I think I used Slingplayer 2.0.4. to originally setup the ProHD originally but when I try to use that version to configure it I get a message about Adobe ActiveX Installer not being found and that seems to have been discontinued. Not sure if there is any mileage in trying to track this down. I've had no joy with that so far.

MY SETUP Humax PVR HDMI output into a HDMI --> Component converter.
Component output from converter --> Slingbox ProHD
ProHD ethernet --> Slinger

Here is the Slinger log below. Not sure where Slinger gets the list of keycodes from BTW
I started up Slinger server followed by VLC for the component output of the ProHD. I then used my remote control to send "power", "pause" and then "1" "0" "4"
__________________________________________________________________________________-

A:\Slinger>cd a:\slinger

A:\Slinger>slingbox_server.exe config_LAN.ini
Version : 3.08f Running on Windows-10-10.0.19044-SP0 pid= 6356 slingbox_server.exe
Using config file config_LAN.ini
Connection Manager Running on port 64000 with 2 max streams using URL sserver.
BOXES [('sb1', 'ProHD_Src1'), ('sb2', 'ProHD_Src2'), ('sb3', 'ProHD_Src3')]
Streamer Running:  2 config_LAN.ini ProHD_Src1 ProHD_Src1 64000 8388608
Streamer Running:  2 config_LAN.ini ProHD_Src2 ProHD_Src2 64000 8388608
Streamer Running:  2 config_LAN.ini ProHD_Src3 ProHD_Src3 64000 8388608
starting up on  port 64000 8388608
Checking for slingbox at ('192.168.0.21', 5001)
Checking for slingbox at ('192.168.0.21', 5001)
Checking for slingbox at ('192.168.0.21', 5001)
('192.168.0.21', 5001) OK
('192.168.0.21', 5001) OK
('192.168.0.21', 5001) OK
ProHD_Src1 Using slingbox at  ('192.168.0.21', 5001)
ProHD_Src2 Using slingbox at  ('192.168.0.21', 5001)
ProHD_Src3 Using slingbox at  ('192.168.0.21', 5001)
Streamer:  ProHD_Src1 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Streamer:  ProHD_Src2 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Streamer:  ProHD_Src3 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Building page for ProHD_Src1
Reading Custom Remote definition from remote_tivo.txt
Building page for ProHD_Src2
Reading Custom Remote definition from remote_humax.txt
Building page for ProHD_Src3
Reading Custom Remote definition from remote_humax.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
12/18/2022, 10:27:49.823   Streaming request from ('127.0.0.1', 50820)

Slinginfo  "Solo/Pro/ProHD" 16 30 192.168.0.21 5001 3000 3
ProHD_Src3 Starting Stream for  127.0.0.1:50820
Connecting... ('192.168.0.21', 5001) Control
ProHD_Src3 Selecting Video Source 3
Keycodes= [1, 1, 2, 3, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 20, 21, 24, 25, 26, 27, 28, 29, 30, 31, 33, 33, 34, 35, 36, 38, 39, 40, 41, 42, 43, 44, 45, 46, 48, 53, 54, 55, 56, 58, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 73, 74, 75, 76, 77, 78, 79, 80, 81]
VideoParameters: Resolution= 16 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 32
Connecting... ('192.168.0.21', 5001) Stream
ProHD_Src3 Stream started at 12/18/2022, 10:27:56.626  733 0
...12/18/2022, 10:28:29.146   RemoteControl connection from ('192.168.0.10', 50840)

Starting remote control stream handler for  ('192.168.0.10', 50840) to port 64001
Remote Control Connected
GET Remote/ProHD_Src3
Exiting Remote Control Stream Handler for ('192.168.0.10', 50840)
12/18/2022, 10:28:29.589   RemoteControl connection from ('192.168.0.10', 50844)

Starting remote control stream handler for  ('192.168.0.10', 50844) to port 64001
Remote Control Connected
GET favicon.ico
Exiting Remote Control Stream Handler for ('192.168.0.10', 50844)
12/18/2022, 10:28:33.591   RemoteControl connection from ('192.168.0.10', 50846)

Starting remote control stream handler for  ('192.168.0.10', 50846) to port 64001
Remote Control Connected
Key Data ('Power', '1')
Key Data ('Digits', '')
Remote Power 1
Exiting Remote Control Stream Handler for ('192.168.0.10', 50846)
ProHD_Src3 Got Streamer Control Message IR
IR [b'\x01192.168.0.10']
12/18/2022, 10:28:33.619  ProHD_Src3 Sending IR keycode 1 3 for 192.168.0.10
sending key 1 3
.12/18/2022, 10:28:39.092   RemoteControl connection from ('192.168.0.10', 50849)

Starting remote control stream handler for  ('192.168.0.10', 50849) to port 64001
Remote Control Connected
Key Data ('Pause', '26')
Key Data ('Digits', '')
Remote Pause 26
Exiting Remote Control Stream Handler for ('192.168.0.10', 50849)
ProHD_Src3 Got Streamer Control Message IR
IR [b'\x1a192.168.0.10']
12/18/2022, 10:28:39.156  ProHD_Src3 Sending IR keycode 26 3 for 192.168.0.10
sending key 26 3
12/18/2022, 10:28:43.395   RemoteControl connection from ('192.168.0.10', 50852)

Starting remote control stream handler for  ('192.168.0.10', 50852) to port 64001
Remote Control Connected
Key Data ('Digits', '')
Key Data ('1', '9')
Remote 1 9
Exiting Remote Control Stream Handler for ('192.168.0.10', 50852)
ProHD_Src3 Got Streamer Control Message IR
IR [b'\t192.168.0.10']
12/18/2022, 10:28:43.419  ProHD_Src3 Sending IR keycode 9 3 for 192.168.0.10
sending key 9 3
12/18/2022, 10:28:46.455   RemoteControl connection from ('192.168.0.10', 50855)

Starting remote control stream handler for  ('192.168.0.10', 50855) to port 64001
Remote Control Connected
Key Data ('Digits', '')
Key Data ('0', '18')
Remote 0 18
Exiting Remote Control Stream Handler for ('192.168.0.10', 50855)
ProHD_Src3 Got Streamer Control Message IR
IR [b'\x12192.168.0.10']
12/18/2022, 10:28:46.523  ProHD_Src3 Sending IR keycode 18 3 for 192.168.0.10
sending key 18 3
.12/18/2022, 10:28:49.099   RemoteControl connection from ('192.168.0.10', 50859)

Starting remote control stream handler for  ('192.168.0.10', 50859) to port 64001
Remote Control Connected
Key Data ('Digits', '')
Key Data ('4', '12')
Remote 4 12
Exiting Remote Control Stream Handler for ('192.168.0.10', 50859)
ProHD_Src3 Got Streamer Control Message IR
IR [b'\x0c192.168.0.10']
12/18/2022, 10:28:49.135  ProHD_Src3 Sending IR keycode 12 3 for 192.168.0.10
sending key 12 3
._________________________________________________________________________________________________________________________----

I'd really appreciate someone having  a look at the remoe control stuff at the end of the log to see if it all looks ok - seems kind of sensible to me.

**Comment by GerryDazoo** [comment]
The keycodes come from the slingbox for the configured video source.
You wrote " I've tested the IR blaster just in case that had stopped working but it works fine on my Tivo box"
But your not using a Tivo box in your setup. You have a Humax PVR connected. The logs look fine, I think the problem you have is the remote setup on the slingbox for the component input is not for the Humax PVR box but something else (Tivo?).
